src/controllers/dynakube/istio/istio_test.go

```diff
@@ -1,128 +1,70 @@
 package istio
 
 import (
-	"encoding/json"
-	"fmt"
 	"net/http"
 	"net/http/httptest"
 	"testing"
 
-	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects"
-	"k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	restclient "k8s.io/client-go/rest"
 )
 
-func initMockServer(t *testing.T, list *metav1.APIGroupList) *httptest.Server {
-	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		var resources any
-		switch req.URL.Path {
-		case "/apis":
-			resources = list
-		default:
-			// t.Logf("unexpected request: %s", req.URL.Path)
-			w.WriteHeader(http.StatusNotFound)
-			return
-		}
-		output, err := json.Marshal(resources)
-		if err != nil {
-			t.Errorf("unexpected encoding error: %v", err)
-			return
-		}
-		w.Header().Set("Content-Type", "application/json")
-		w.WriteHeader(http.StatusOK)
-		_, _ = w.Write(output)
-	}))
-
-	return server
-}
-
 func TestIstioEnabled(t *testing.T) {
-	// resource is enabled
-	list := &metav1.APIGroupList{
-		Groups: []metav1.APIGroup{
-			{
-				Name: "networking.istio.io",
-				Versions: []metav1.GroupVersionForDiscovery{
-					{GroupVersion: "v1alpha3", Version: "v1alpha3"},
-					{GroupVersion: "v1alpha3", Version: "v1alpha3"},
-				},
-			},
-		},
-	}
-	server := initMockServer(t, list)
+	server := initMockServer(true)
 	defer server.Close()
 	cfg := &restclient.Config{Host: server.URL}
-	r, e := CheckIstioEnabled(cfg)
+	r, e := CheckIstioInstalled(cfg)
 	if r != true {
 		t.Error(e)
 	}
 }
 
 func TestIstioDisabled(t *testing.T) {
-	// resource is not enabled
-	list := &metav1.APIGroupList{
-		Groups: []metav1.APIGroup{
-			{
-				Name: "not.istio.group",
-				Versions: []metav1.GroupVersionForDiscovery{
-					{GroupVersion: "v1alpha3", Version: "v1alpha3"},
-					{GroupVersion: "v1alpha3", Version: "v1alpha3"},
-				},
-			},
-		},
-	}
-	server := initMockServer(t, list)
+	server := initMockServer(false)
 	defer server.Close()
 	cfg := &restclient.Config{Host: server.URL}
-	r, e := CheckIstioEnabled(cfg)
+	r, e := CheckIstioInstalled(cfg)
 	if r != false && e == nil {
 		t.Errorf("expected false, got true, %v", e)
 	}
```

Comment:
same as before, using `require` or `assert` (we have to improve the old code at some point 😅 )